tests/nosetests/pyanaconda_tests/module_subscription_tests.py

```diff
@@ -0,0 +1,311 @@
+        )
+
+    def kickstart_properties_test(self):
+        """Test kickstart properties."""
+        self.assertEqual(self.subscription_interface.KickstartCommands, ["syspurpose", "rhsm"])
+        self.assertEqual(self.subscription_interface.KickstartSections, [])
+        self.assertEqual(self.subscription_interface.KickstartAddons, [])
+        self.callback.assert_not_called()
+
+    def system_purpose_data_test(self):
+        """Test the SystemPurposeData DBus structure."""
+
+        # create the SystemPurposeData structure
+        system_purpose_data = SystemPurposeData()
+        system_purpose_data.role = "foo"
+        system_purpose_data.sla = "bar"
+        system_purpose_data.usage = "baz"
+        system_purpose_data.addons = ["a", "b", "c"]
+
+        # create its expected representation
+        expected_dict = {
+            "role": get_variant(Str, "foo"),
+            "sla": get_variant(Str, "bar"),
+            "usage": get_variant(Str, "baz"),
+            "addons": get_variant(List[Str], ["a", "b", "c"])
+        }
+
+        # compare the two
+        self.assertEqual(SystemPurposeData.to_structure(system_purpose_data), expected_dict)
+
+    def set_system_purpose_test(self):
+        """Test if setting system purpose data from DBUS works correctly."""
+        system_purpose_data = {
+            "role": get_variant(Str, "foo"),
+            "sla": get_variant(Str, "bar"),
+            "usage": get_variant(Str, "baz"),
+            "addons": get_variant(List[Str], ["a", "b", "c"])
+        }
+
+        self._test_dbus_property(
+          "SystemPurposeData",
+          system_purpose_data
+        )
+
+        output_structure = self.subscription_interface.SystemPurposeData
+        output_system_purpose_data = SystemPurposeData.from_structure(output_structure)
+
+        self.assertEqual(output_system_purpose_data.role, "foo")
+        self.assertEqual(output_system_purpose_data.sla, "bar")
+        self.assertEqual(output_system_purpose_data.usage, "baz")
+        self.assertEqual(output_system_purpose_data.addons, ["a", "b", "c"])
+
+    def ks_set_all_test(self):
+        """Test if setting all options from kickstart works correctly."""
+        self.subscription_interface.ReadKickstart(
+            'syspurpose --role="FOO" --sla="BAR" --usage="BAZ" '
+            '--addon="Foo Product" --addon="Bar Feature"'
+        )
+        structure = self.subscription_interface.SystemPurposeData
+        system_purpose_data = SystemPurposeData.from_structure(structure)
+        self.assertEqual(system_purpose_data.role, 'FOO')
+        self.assertEqual(system_purpose_data.sla, 'BAR')
+        self.assertEqual(system_purpose_data.usage, 'BAZ')
+        self.assertEqual(system_purpose_data.addons, ["Foo Product", "Bar Feature"])
+
+    def ks_set_nothing_test(self):
+        """Test what happens if just the syspurpose command is used."""
+        self.subscription_interface.ReadKickstart('syspurpose')
```

Comment:
Please, merge this test with `ks_out_command_only_test`. It is not a good practice to call `ReadKickstart` directly in the tests. You would have to check the return value to make sure that the parsing was sucesfull.

Reply:
Good point! Looks like I have not realized the `check_kickstart_interface` function also changes module state, so I can check it, not just the output kickstart. That simplifies things quite a bit, thanks! :)